plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -165,28 +222,30 @@ instance ExMemoryUsage Integer where
 instance ExMemoryUsage BS.ByteString where
   memoryUsage bs = ExMemory $ ((n-1) `quot` 8) + 1  -- Don't use `div` here!  That gives 1 instead of 0 for n=0.
       where n = fromIntegral $ BS.length bs :: SatInt
+  {-# INLINE memoryUsage #-}
 
 instance ExMemoryUsage T.Text where
   -- This is slow and inaccurate, but matches the version that was originally deployed.
   -- We may try and improve this in future so long as the new version matches this exactly.
   memoryUsage text = memoryUsage $ T.unpack text
+  {-# INLINE memoryUsage #-}
 
 instance ExMemoryUsage Int where
   memoryUsage _ = 1
+  {-# INLINE memoryUsage #-}
 
 instance ExMemoryUsage Char where
   memoryUsage _ = 1
+  {-# INLINE memoryUsage #-}
 
 instance ExMemoryUsage Bool where
   memoryUsage _ = 1
+  {-# INLINE memoryUsage #-}
 
 -- Memory usage for lists: let's just go for a naive traversal for now.
 instance ExMemoryUsage a => ExMemoryUsage [a] where
-    memoryUsage = sizeList
-        where sizeList =
-                  \case
-                   []   -> 0
-                   x:xs -> memoryUsage x + sizeList xs
+    memoryUsage = foldl' (\a x -> memoryUsage x + a) 0
```

Comment:
If you're going to do this you should change/delete the comment before it.

Reply:
I thought this still was a naive traversal (compared to storing the cost alongside the list or whatever). Ok, will fix.